pkg/apis/apps/validation/validation_test.go

```diff
@@ -790,148 +790,90 @@ func TestValidateStatefulSetMinReadySeconds(t *testing.T) {
-		updatedReplicas    int32
-		availableReplicas  int32
-		observedGeneration *int64
-		collisionCount     *int32
-		expectedErr        bool
-	}{
-		{
-			name:            "valid status",
-			replicas:        3,
-			readyReplicas:   3,
-			currentReplicas: 2,
-			updatedReplicas: 1,
-			expectedErr:     false,
-		},
-		{
-			name:            "invalid replicas",
-			replicas:        -1,
-			readyReplicas:   3,
-			currentReplicas: 2,
-			updatedReplicas: 1,
-			expectedErr:     true,
-		},
-		{
-			name:            "invalid readyReplicas",
-			replicas:        3,
-			readyReplicas:   -1,
-			currentReplicas: 2,
-			updatedReplicas: 1,
-			expectedErr:     true,
-		},
-		{
-			name:            "invalid currentReplicas",
-			replicas:        3,
-			readyReplicas:   3,
-			currentReplicas: -1,
-			updatedReplicas: 1,
-			expectedErr:     true,
-		},
-		{
-			name:            "invalid updatedReplicas",
-			replicas:        3,
-			readyReplicas:   3,
-			currentReplicas: 2,
-			updatedReplicas: -1,
-			expectedErr:     true,
-		},
-		{
-			name:               "invalid observedGeneration",
-			replicas:           3,
-			readyReplicas:      3,
-			currentReplicas:    2,
-			updatedReplicas:    1,
-			observedGeneration: &observedGenerationMinusOne,
-			expectedErr:        true,
-		},
-		{
-			name:            "invalid collisionCount",
-			replicas:        3,
-			readyReplicas:   3,
-			currentReplicas: 2,
-			updatedReplicas: 1,
-			collisionCount:  &collisionCountMinusOne,
-			expectedErr:     true,
-		},
-		{
-			name:            "readyReplicas greater than replicas",
-			replicas:        3,
-			readyReplicas:   4,
-			currentReplicas: 2,
-			updatedReplicas: 1,
-			expectedErr:     true,
-		},
-		{
-			name:            "currentReplicas greater than replicas",
-			replicas:        3,
-			readyReplicas:   3,
-			currentReplicas: 4,
-			updatedReplicas: 1,
-			expectedErr:     true,
-		},
-		{
-			name:            "updatedReplicas greater than replicas",
-			replicas:        3,
-			readyReplicas:   3,
-			currentReplicas: 2,
-			updatedReplicas: 4,
-			expectedErr:     true,
-		},
-		{
-			name:              "invalid: number of available replicas",
-			replicas:          3,
-			readyReplicas:     3,
-			currentReplicas:   2,
-			availableReplicas: int32(-1),
-			expectedErr:       true,
-		},
-		{
-			name:              "invalid: available replicas greater than replicas",
-			replicas:          3,
-			readyReplicas:     3,
-			currentReplicas:   2,
-			availableReplicas: int32(4),
-			expectedErr:       true,
-		},
-		{
-			name:              "invalid: available replicas greater than ready replicas",
-			replicas:          3,
-			readyReplicas:     2,
-			currentReplicas:   2,
-			availableReplicas: int32(3),
-			expectedErr:       true,
-		},
+func validateStatefulSetStatus(replicas int32, readyReplicas int32, currentReplicas int32, updatedReplicas int32, availableReplicas int32, observedGeneration *int64, collisionCount *int32) error {
```

Comment:
```suggestion
func validateStatefulSetStatus(replicas, readyReplicas, currentReplicas, updatedReplicas, availableReplicas int32, observedGeneration *int64, collisionCount *int32) error {
```

To be honest this function is already pretty long.  I may suggest refactoring this into a struct so its a bit more readable.

Reply:
Sounds good! Will do!